lib/engine/corporation.rb

```diff
@@ -92,6 +92,7 @@ def buy_multiple?
     def can_par?(entity)
       return false if @par_via_exchange && @par_via_exchange.owner != entity
       return false if @needs_token_to_par && @tokens.empty?
+      return false if all_abilities.find { |a| a.type == :unparrable }
```

Comment:
it might be easier to move can_par? into the game class

Reply:
`can_par?` in `Corporation` now calls `corporation_parrable?` in `Game` so that `SharePool`'s `can_par?` can be differentiated later as `parrable_at?` or something like that if needed in the future

Reply:
This broke pretty thoroughly; `Corporation` doesn't have a reference to `@game`. 
I think there are three main solutions to this:
1: remove `can_par?` from corporation and just use a method in `game`
 - This is doable without much effort, I'd want to give heads up to the 1824 and 1849 developers that this will slightly affect their games
 2: add `@game` to `Corporation`
  - This seems like it would be contrary to the goal of avoiding packing logic in `Corporation`
 3: revert to how it was in the PR originally
 
 I'm fine with any direction on this, @tobymao do you have any preferences?

Reply:
i don't know what version 3 is but i was suggesting 1.

entities shouldn't require game objects, if they do, it's a smell because it's game -> entity -> game, by moving it to game, yes game gets bigger, but it's just much easier to override etc.

entities should be mostly simple data container objects without business logic. because if there's logic, then a game can and will override it